lib/active_model/serializer.rb

```diff
@@ -102,7 +111,16 @@ def self.adapter
     end
 
     def self.root_name
-      name.demodulize.underscore.sub(/_serializer$/, '') if name
+      name.demodulize.underscore.sub(/_serialization$/, '') if name
+    end
+
+    def self.deserialize(params)
+      association_keys = _reflections.map(&:name)
+      permitted_params = adapter.params(@_params, association_keys) || @_params
+      permitted_key    = adapter.root || root_name
+
+      sanitazed_params = params.require(permitted_key.to_sym).permit(permitted_params)
+      adapter.parse(sanitazed_params)
```

Comment:
Now that #954 is merged, I imagine this method should be moved into `ActiveModel::SerializableResource.deserialize(resource, options)` to parallel https://github.com/bf4/active_model_serializers/blob/df140293d3010ecbbac50061c22e03a154d1a761/lib/active_model/serializable_resource.rb#L15-L27